### ISSUE TITLE
`exactOptionalPropertyTypes` type issues

### ISSUE THREAD

**Issue by macmillen** [sveltejs/language-tools#1176]
**Describe the bug**
When the `exactOptionalPropertyTypes` flag is set to true in `tsconfig.json` I get errors when passing values to HTML elements which are possibly `undefined` which (presumably) shouldn't be happening.

**To Reproduce**

1. **Example:**
``` svelte
<button on:click>Button</button>
```

```
Type 'undefined' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.ts(2322)
(JSX attribute) svelte.JSX.DOMAttributes<HTMLButtonElement>.onclick?: svelte.JSX.MouseEventHandler<HTMLButtonElement>
```

2. **Example**
``` svelte
<script lang="ts">
  let placeholder: string | undefined;
</script>

<input {placeholder} />
```

```
Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.ts(2322)
```

**Expected behavior**
All HTML properties should accept an `undefined` value. (?)

**Screenshots**
![image](https://user-images.githubusercontent.com/33156526/133223489-ea7d0433-8e4c-41d0-8e36-45ca333e8d5b.png)

**System (please complete the following information):**
 - OS: Linux
 - IDE: VSCode
 - Plugin/Package: Svelte for VSCode

**Comment by dummdidumm** [comment]
Svelte will treat `undefined` as "remove that attribute", so yeah, all attributes should have the `undefined` value. Event listeners is a different thing, in this case the transformation probably needs to be adjusted. I'm hesitant to allow undefined as an event listener. It's technically possible though.

**Comment by macmillen** [comment]
@dummdidumm, thanks for the fix! The error messages for the event listeners are still a problem though. Should we leave this issue open or create a new one?

**Comment by dummdidumm** [comment]
Oh I thought that's fixed, too, I'll reopen an look into it.

**Comment by macmillen** [comment]
Yes I meant the forwarded events. It seems in VSCode the errors don't show up anymore but when running `svelte-check` i still get this for example:
```
Error: Type 'undefined' is not assignable to type 'FocusEventHandler<HTMLTextAreaElement>'. (ts)
```

---


And there is also a problem with transitions when the function can be `undefined`. In the following snippet you can see that the `crossfade` function returns the `send` and `receive` transition functions which can be `undefined` which results in an error.

It would be nice to be able to pass `undefined` because then you could pass down optional transitions as component props as well for example if you not always want to have a transition for a specific component.

``` svelte
<script lang="ts">
  import { crossfade } from "svelte/transition";

  const [send, receive] = crossfade({ duration: 400 });
</script>

<div
  in:receive={{ key }}
  out:send={{ key }}
/>
```

```
Cannot invoke an object which is possibly 'undefined'.ts(2722)
```
